src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs

```diff
@@ -6548,28 +6548,59 @@ internal sealed partial class LazyTranslatedQueryExpression : BaseTranslatedQuer
         protected override IOperation ExpressionImpl => _lazyExpression.Value;
     }
 
-    internal sealed partial class FlowCapture : Operation, IFlowCaptureOperation
+    internal sealed partial class FlowCaptureReference : Operation, IFlowCaptureReferenceOperation
     {
-        public FlowCapture(int id, SyntaxNode syntax, bool isInitialization, ITypeSymbol type, Optional<object> constantValue) :
-            base(OperationKind.FlowCapture, semanticModel: null, syntax, type, constantValue, isImplicit: true)
+        public FlowCaptureReference(int id, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue) :
+            base(OperationKind.FlowCaptureReference, semanticModel: null, syntax, type, constantValue, isImplicit: true)
         {
             Id = id;
-            IsInitialization = isInitialization;
         }
 
         public int Id { get; }
-        public bool IsInitialization { get; }
         public override IEnumerable<IOperation> Children
         {
             get
             {
                 return Array.Empty<IOperation>();
             }
         }
+
+        public override void Accept(OperationVisitor visitor)
+        {
+            visitor.VisitFlowCaptureReference(this);
+        }
+
+        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
+        {
+            return visitor.VisitFlowCaptureReference(this, argument);
+        }
+    }
+
+    internal sealed partial class FlowCapture : Operation, IFlowCaptureOperation
+    {
+        public FlowCapture(int id, SyntaxNode syntax, IOperation value) :
+            base(OperationKind.FlowCapture, semanticModel: null, syntax, type: null, constantValue: default, isImplicit: true)
+        {
+            Debug.Assert(value != null);
+            Id = id;
+            Value = Operation.SetParentOperation(value, this);
```

Comment:
This approach is different from other nodes, where we lazily set the `Parent` only when the consumer requests the parent node. Is there a specific reason we set the parent in the constructor for these nodes?

Reply:
> This approach is different from other nodes, where we lazily set the Parent only when the consumer requests the parent node. Is there a specific reason we set the parent in the constructor for these nodes?

There is nothing lazy about the nodes, so all that lazy parent infrastructure is just an overhead. It also prevents us from giving out children on their own because the lazy parent often requires us to always reach to the child from its parent, otherwise we could be in trouble. So, I think we should get rid of all that lazy parent machinery for non-lazy flavor of IOperation nodes

---
In reply to: [163289663](https://github.com/dotnet/roslyn/pull/24358#discussion_r163289663) [](ancestors = 163289663)